source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -1393,6 +1395,29 @@ bool ServerContextImpl::TlsContext::isCipherEnabled(uint16_t cipher_id, uint16_t
   return false;
 }
 
+bool ContextImpl::verifyCertChain(bssl::UniquePtr<X509> leaf_cert,
```

Comment:
Same on these two params: I don't think they should take ownership. Pass a `&` or `const &`. Unless I'm missing a subtlety of the boringssl API and ownership.